code/modules/mob/living/simple_animal/borer/borer.dm

```diff
@@ -0,0 +1,1023 @@
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 1500
+
+	var/generation = 1
+	var/static/list/borer_names = list(
+			"Primary", "Secondary", "Tertiary", "Quaternary", "Quinary", "Senary",
+			"Septenary", "Octonary", "Novenary", "Decenary", "Undenary", "Duodenary",
+			)
+
+	var/mob/living/carbon/victim = null
+	var/mob/living/captive_brain/host_brain = null
+	var/truename
+	var/docile = 0
+	var/bonding = FALSE
+	var/controlling = 0
+	var/chemicals = 10
+	var/used_dominate
+	var/borer_chems = list()
+	var/dominate_cooldown = 150
+	var/leaving = 0
+	var/hiding = FALSE
+
+	var/datum/action/innate/borer/talk_to_host/talk_to_host_action = new
+	var/datum/action/innate/borer/infest_host/infest_host_action = new
+	var/datum/action/innate/borer/toggle_hide/toggle_hide_action = new
+	var/datum/action/innate/borer/talk_to_borer/talk_to_borer_action = new
+	var/datum/action/innate/borer/talk_to_brain/talk_to_brain_action = new
+	var/datum/action/innate/borer/take_control/take_control_action = new
+	var/datum/action/innate/borer/give_back_control/give_back_control_action = new
+	var/datum/action/innate/borer/leave_body/leave_body_action = new
+	var/datum/action/innate/borer/make_chems/make_chems_action = new
+	var/datum/action/innate/borer/make_larvae/make_larvae_action = new
+	var/datum/action/innate/borer/freeze_victim/freeze_victim_action = new
+	var/datum/action/innate/borer/punish_victim/punish_victim_action = new
+	var/datum/action/innate/borer/jumpstart_host/jumpstart_host_action = new
+
+/mob/living/simple_animal/borer/New(atom/newloc, var/gen=1)
+	..(newloc)
+	generation = gen
+	notify_ghosts("A cortical borer has been created in [get_area(src)]!", enter_link = "<a href=?src=\ref[src];ghostjoin=1>(Click to enter)</a>", source = src, action = NOTIFY_ATTACK)
+	real_name = "Cortical Borer [rand(1000,9999)]"
+	truename = "[borer_names[min(generation, borer_names.len)]] [rand(1000,9999)]"
+	borer_chems += /datum/borer_chem/epinephrine
+	borer_chems += /datum/borer_chem/leporazine
+	borer_chems += /datum/borer_chem/mannitol
+	borer_chems += /datum/borer_chem/bicaridine
+	borer_chems += /datum/borer_chem/kelotane
+	borer_chems += /datum/borer_chem/charcoal
+	borer_chems += /datum/borer_chem/methamphetamine
+	borer_chems += /datum/borer_chem/salbutamol
+	borer_chems += /datum/borer_chem/spacedrugs
+	//borer_chems += /datum/borer_chem/creagent
+	borer_chems += /datum/borer_chem/ethanol
+	borer_chems += /datum/borer_chem/rezadone
+
+	borers += src
+
+	GrantBorerActions()
+
+/mob/living/simple_animal/borer/attack_ghost(mob/user)
+	if(jobban_isbanned(user, ROLE_BORER) || jobban_isbanned(user, "Syndicate"))
+		return
+	if(ckey)
+		return
+	if(stat != CONSCIOUS)
+		return
+	var/be_swarmer = alert("Become a cortical borer? (Warning, You can no longer be cloned!)",,"Yes","No")
+	if(be_swarmer == "No")
+		return
+	if(qdeleted(src))
```

Comment:
no need to have this if you check it right below, it only needs to be checked once after sleeps